### PR TITLE
[CALCITE-4801] SubstitutionVisitor support query pull up no-equal filter, when mv's aggregate rollup to query's aggregate(xurenhe)

### PR DESCRIPTION
**Pull request by wojustme**
Enhance SubstitutionVisitor, when query pull-up filter under aggregate, as [CALCITE-4801](https://issues.apache.org/jira/browse/CALCITE-4801) described.

### CODE REVIEW

**Comment by wojustme** [comment]
Hi @hsyuan @yanlin-Lynn @xy2953396112 
Please review this pr, if you have time.
Thanks a lot.

File: core/src/test/java/org/apache/calcite/test/MaterializedViewSubstitutionVisitorTest.java [hunk 1]
```diff
@@ -1041,6 +1041,77 @@
...
+        + "from \"emps\" group by \"name\"";
+    final String query = ""
+        + "select count(distinct \"deptno\") as cnt\n"
+        + "from \"emps\" where \"name\" <> 'hello'";
+    sql(mv, query).withChecker(
+        resultContains(""
+            + "LogicalCalc(expr#0..1=[{inputs}], expr#2=['hello':VARCHAR], expr#3=[CAST($t0):VARCHAR], "
+            + "expr#4=[<>($t2, $t3)], CNT=[$t1], $condition=[$t4])\n"
+            + "  EnumerableTableScan(table=[[hr, MV0]])")).ok();
+  }
```
Comment (yanlin-Lynn) [inline core/src/test/java/org/apache/calcite/test/MaterializedViewSubstitutionVisitorTest.java:1057]: this is not correct!

Comment (wojustme) [reply core/src/test/java/org/apache/calcite/test/MaterializedViewSubstitutionVisitorTest.java:1057]: Ok, it's really invalid.
I will close it, thanks for your review.

**Comment by wojustme** [comment]
Close it because of invalid [issue-4801](https://issues.apache.org/jira/browse/CALCITE-4801).

**Comment by wojustme** [comment]
Close it because of invalid [issue-4801](https://issues.apache.org/jira/browse/CALCITE-4801).

**Pull request closed without merging**